Cannot build Maestro using Java 21

### Is there an existing issue for this?

- [x] I have searched the existing issues and didn't find mine.

### Steps to reproduce

./installLocally.sh

### Actual results

```
dan@dan-thinkbook:~/git/maestro/maestro$ ./installLocally.sh 
> Task :maestro-studio:server:compileKotlin FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':maestro-studio:server:compileKotlin'.
> ⛔ Inconsistent JVM Target Compatibility Between Java and Kotlin Tasks
  Inconsistent JVM-target compatibility detected for tasks 'compileJava' (17) and 'compileKotlin' (21).
  Solution: Consider using JVM Toolchain: https://kotl.in/gradle/jvm/toolchain
  Learn more about JVM-target validation: https://kotl.in/gradle/jvm/target-validation

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.5/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
28 actionable tasks: 1 executed, 27 up-to-date
cp: cannot stat './maestro-cli/build/install/maestro/bin': No such file or directory
cp: cannot stat './maestro-cli/build/install/maestro/lib': No such file or directory


dan@dan-thinkbook:~/git/maestro/maestro$ java -version
openjdk version "21.0.8" 2025-07-15 LTS
OpenJDK Runtime Environment Temurin-21.0.8+9 (build 21.0.8+9-LTS)
OpenJDK 64-Bit Server VM Temurin-21.0.8+9 (build 21.0.8+9-LTS, mixed mode, sharing)
```

### Expected results

Build and install

### About app

n/a

### About environment

Linux using Java 21.0.8-tem installed via sdkman

### Logs

n/a

### Maestro version

main

### How did you install Maestro?

built from source (please include commit hash in the text area below)

### Anything else?

Switching to 17.0.6-tem worked fine. Suspect there's just a version declaration missing somewhere.

--- a/maestro-cli/build.gradle.kts
+++ b/maestro-cli/build.gradle.kts
@@ -209,6 +209,10 @@ java {
     targetCompatibility = JavaVersion.VERSION_17
 }
 
+kotlin {
+    jvmToolchain(17)
+}
+
 tasks.named("compileKotlin", KotlinCompilationTask::class.java) {
     compilerOptions {
         freeCompilerArgs.addAll("-Xjdk-release=17")

--- a/maestro-studio/server/build.gradle
+++ b/maestro-studio/server/build.gradle
@@ -1,4 +1,3 @@
-import com.vanniktech.maven.publish.SonatypeHost
 import org.jetbrains.kotlin.gradle.tasks.KotlinCompilationTask
 
 plugins {
@@ -41,6 +40,10 @@ tasks.whenTaskAdded {
 java {
     sourceCompatibility = JavaVersion.VERSION_17
     targetCompatibility = JavaVersion.VERSION_17
+}
+
+kotlin {
+    jvmToolchain(17)
 }
 
 tasks.named("compileKotlin", KotlinCompilationTask) {

